charmcraft/commands/_utils.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+import os
+from stat import S_IXUSR, S_IXGRP, S_IXOTH
+
+S_IXALL = S_IXUSR | S_IXGRP | S_IXOTH
+
+
+def make_executable(fh):
+    """make open file fh executable"""
+    fileno = fh.fileno()
+    os.fchmod(fileno, os.fstat(fileno).st_mode | S_IXALL)
```

Comment:
usually you only set X for ones that have read/write something else. I'm ok with this but it does seem odd that you might set executable when that person cannot read the content.
If you wanted that I think it is:
```
mode = os.fstat(fileno).st_mode
mode_r = mode & (S_IRUSR | S_IRGRP | S_IROTH)
mode_x = mode_r >> 2
mode = mode | mode_x
```